47.md

```diff
@@ -135,6 +137,95 @@ The **client** should then store this connection and use it when the user wants
 nostr+walletconnect://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&secret=71a8c14c1407c113601079c4302dab36460f0ccd0ad506f1f2dc73b5100e4f3c
 ```
 
+## Client-Created Secret
+
+The client MAY generate its own secret and co-ordinate only public keys with the wallet service so that secrets are never exposed.
+
+### HTTP Confirmation Page
+
+A **wallet service** MAY host a publicly accessible confirmation page that the **client** may open (e.g. in a webview or popup window) and allow the **user** to confirm the creation of a new connection, with the following parameters:
+
+- `pubkey` Required. The corresponding pubkey of the secret generated by the **client**.
+
+The **wallet service** MAY ignore all the below optional parameters.
+
+- `name`: Optional. The name of the client app.
+- `icon`: Optional. The URL of an icon of the client app to display on the confirmation page.
+- `return_to` Optional. URI to open after the connection is created. The **wallet service** SHOULD add the `relay_url`, `wallet_pubkey` and `lud16` parameters to the URI.
+- `expires_at` Optional. The connection cannot be used after this date. Unix timestamp in seconds.
+- `max_amount` Optional. The maximum amount in millisats that can be sent per renewal period
+- `budget_renewal` Optional. The reset the budget at the end of the given budget renewal. Can be `never` (default), `daily`, `weekly`, `monthly`, `yearly`
+- `request_methods` Optional. The url encoded, space separated list of request types that you need permission for. For example: `..&request_methods=pay_invoice%20get_balance`
+- `notification_types` Optional. The url encoded, space separated list of notification types that you need permission for. For example: `..&notification_types=payment_received%20payment_sent`
+- `isolated` Optional. The makes an isolated app connection / sub-wallet with its own balance and only access to its own transaction list. e.g. `&isolated=true`.
```

Comment:
I am not sure what this means. Did I miss some development regarding NWC? The description sounds like how I would expect a NWC connection to behave isn't the default so I am not sure what the default without `isolated` then is.

As-is, as an app developer, I wouldn't know if I should add `&isolated=true` or not.

Reply:
Currently this is only implemented in Alby Hub, but I think it's a powerful feature which is why I'm proposing it as an optional parameter.

There are different usecases for isolated and non-isolated connections.

If you need access to all wallet activity or want a standard budgeted connection without having to top it up, you do not want an isolated connection.

If you wanted to create a PoS for a specific product/event/business, or sub-wallet for a friend/family member, you probably want an isolated connection, so that they can only see their own sent/received payments. Isolated connections (sub-wallets) start with 0 balance and must be topped up before they are spendable.

Reply:
> Isolated connections (sub-wallets) start with 0 balance and must be topped up before they are spendable.

So if `isolated` is set, `budget_renewal` and `max_amount` don't make sense, right?

I think that's okay if that's the case, but I think we need to explain `isolated` more in this spec else it's confusing.

Just from reading the current text, I would erroneously think I want to use `isolated`.

I don't have suggestions to improve right now though. I mean, I get it now, but I am not sure about the next guy haha

Reply:
@ekzyis in most cases `budget_renewal` and `max_amount` don't make sense for isolated connections, but there are still use cases though: e.g. a sub-wallet you give to your kids but only allow them to spend some of their money each month to incentivize saving part of their pocket money (just as a concrete example).